src/main/scala/wdl4s/expression/TypeEvaluator.scala

```diff
@@ -91,6 +91,11 @@ case class TypeEvaluator(override val lookup: String => WdlType, override val fu
         case (_, Failure(ex)) => Failure(ex)
         case (_, _) => Failure(new WdlExpressionException(s"Can't index ${a.toPrettyString}"))
       }
+    case a: Ast if a.isTernaryOperator =>
+      (evaluate(a.getAttribute("true")), evaluate(a.getAttribute("false"))) match {
+        case (Success(l), Success(r)) if l == r => Success(l)
+        case (_, _) => Failure(new WdlExpressionException(s"True and false paths of ternary operator do not evaluate to the same type"))
```

Comment:
Good catch... I'm not sure how to handle coercion, maybe we can't type-evaluate ternary operators statically if each branch is allowed to return a different type